lib/rubocop/cop/rails/root_pathname_methods.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use `Rails.root` IO methods instead of passing it to `File`.
+      #
+      # `Rails.root` is an instance of `Pathname`
+      # so we can apply many IO methods directly.
+      #
+      # This cop works best when used together with
+      # `Style/FileRead`, `Style/FileWrite` and `Rails/RootJoinChain`.
+      #
+      # @example
+      #   # bad
+      #   File.open(Rails.root.join('db', 'schema.rb'))
+      #   File.open(Rails.root.join('db', 'schema.rb'), 'w')
+      #   File.read(Rails.root.join('db', 'schema.rb'))
+      #   File.binread(Rails.root.join('db', 'schema.rb'))
+      #   File.write(Rails.root.join('db', 'schema.rb'), content)
+      #   File.binwrite(Rails.root.join('db', 'schema.rb'), content)
+      #
+      #   # good
+      #   Rails.root.join('db', 'schema.rb').open
+      #   Rails.root.join('db', 'schema.rb').open('w')
+      #   Rails.root.join('db', 'schema.rb').read
+      #   Rails.root.join('db', 'schema.rb').binread
+      #   Rails.root.join('db', 'schema.rb').write(content)
+      #   Rails.root.join('db', 'schema.rb').binwrite(content)
+      #
+      class RootPathnameMethods < Base
+        extend AutoCorrector
+
+        MSG = '`%<rails_root>s` is a `Pathname` so you can just append `#%<method>s`.'
+
+        DIR_METHODS = %i[
+          children
+          delete
+          each_child
+          empty?
+          entries
+          exist?
+          glob
+          mkdir
+          open
+          rmdir
+          unlink
+        ].to_set.freeze
+
+        FILE_METHODS = %i[
```

Comment:
`File.methods - File.superclass.methods` also returns e.g. `:absolute_path` which is not on those lists.
Which makes me think does it make sense to fill in those constants with those calls?
On the other hand, if the Ruby version with which `rubocop` is run is different from the target Ruby version, there might be differences.
However, does it matter much?

Reply:
... but `Pathname` does not have a `#absolute_path` method. Not sure I can follow here.

I generated this list using

```ruby
File.methods.sort.select do |m|
  Pathname.instance_methods.include?(m) && !Object.instance_methods.include?(m)
end
```

Reply:
Understood.
So basically `(File.public_methods(false) & Pathname.instance_methods(false)).sort`.
Does it make sense to set the constant to this statement?
What are the consequences? New methods being added on `File` and to `Pathname` in future Ruby versions change the behaviour of the cop?

Reply:
Well, just because there is `File.foo` and `Pathname#foo`, it is not guaranteed that `File.foo(p, ...)` and `p.foo(...)` is the same, so I'd rather maintain a list of safe methods.

Also, there may be cases where the Ruby that runs Rubocop is not the Ruby which runs production code (e.g. you develop a gem on Ruby 2, but someone uses it on JRuby 9). Are those standard libraries necessarily equivalent?